server/src/main/java/org/opensearch/extensions/rest/RestInitializeExtensionAction.java

```diff
@@ -0,0 +1,122 @@
+        String name = "";
+        String uniqueId = "";
+        String hostAddress = "";
+        String port = "";
+        String version = "";
+        String openSearchVersion = "";
+        String minimumCompatibleVersion = "";
+        List<ExtensionDependency> dependencies = new ArrayList<>();
+
+        if (request.hasContent()) {
+            try (XContentParser parser = request.contentParser()) {
+                parser.nextToken();
+                ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+                while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+                    String currentFieldName = parser.currentName();
+                    parser.nextToken();
+                    if ("name".equals(currentFieldName)) {
+                        name = parser.text();
+                    } else if ("uniqueId".equals(currentFieldName)) {
+                        uniqueId = parser.text();
+                    } else if ("hostAddress".equals(currentFieldName)) {
+                        hostAddress = parser.text();
+                    } else if ("port".equals(currentFieldName)) {
+                        port = parser.text();
+                    } else if ("version".equals(currentFieldName)) {
+                        version = parser.text();
+                    } else if ("opensearchVersion".equals(currentFieldName)) {
+                        openSearchVersion = parser.text();
+                    } else if ("minimumCompatibleVersion".equals(currentFieldName)) {
+                        minimumCompatibleVersion = parser.text();
+                    } else if ("dependencies".equals(currentFieldName)) {
+                        ensureExpectedToken(XContentParser.Token.START_ARRAY, parser.currentToken(), parser);
+                        while (parser.nextToken() != XContentParser.Token.END_ARRAY) {
+                            dependencies.add(ExtensionDependency.parse(parser));
+                        }
+                    }
+                }
+            }
+        }
+
+        Extension extension = new Extension(
+            name,
+            uniqueId,
+            hostAddress,
+            port,
+            version,
+            openSearchVersion,
+            minimumCompatibleVersion,
+            dependencies,
+            // TODO create parser for additionalSettings https://github.com/opensearch-project/OpenSearch/issues/8032
+            null
+        );
+        try {
+            extensionsManager.loadExtension(extension);
+            extensionsManager.initialize();
+        } catch (IOException e) {
+            return channel -> channel.sendResponse(new BytesRestResponse(RestStatus.INTERNAL_SERVER_ERROR, e.getMessage()));
+
+        }
+
+        return channel -> {
+            try (XContentBuilder builder = channel.newBuilder()) {
+                builder.startObject();
+                builder.field("Extension has been initialized");
```

Comment:
You need both a field name and content, otherwise this will generate invalid JSON.
```suggestion
                builder.field("message", "Extension has been initialized");
```